distributed/cli/tests/test_dask_scheduler.py

```diff
@@ -255,3 +254,49 @@ def test_bokeh_port_zero(loop):
                 if b'bokeh' in line.lower() or b'web' in line.lower():
                     count += 1
                     assert b':0' not in line
+
+
+PRELOAD_TEXT = """
+from distributed.scheduler import Scheduler
+
+def dask_setup(scheduler):
+    if isinstance(scheduler, Scheduler):
+        print("setup")
+
+def dask_teardown(scheduler):
+    if isinstance(scheduler, Scheduler):
+        print("teardown")
+"""
+
+
+def test_preload_file(loop):
```

Comment:
I don't know precisely, but I'm not surprised.  Windows handles things differently.  Maybe a file needs to be flushed?  My recommendation is to avoid this problem by staying within Python and away from the OS.  For example you might attach a new attribute/flag to the scheduler and then run a function on the scheduler from the client to verify that it has arrived

```python
def f(dask_scheduler):
    assert dask_scheduler.my_flag

c.run_on_scheduler(f)
```